python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9059,3 +9059,54 @@ def test_dataframe_add_suffix():
     expected = pdf.add_suffix("_item")
 
     assert_eq(got, expected)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.normal(-100, 100, (50, 50)),
+        np.random.randint(-50, 50, (25, 30)),
+        np.random.random_sample((4, 4)),
+        np.random.uniform(10.5, 75.5, (10, 6)),
+        np.array([1.123, 2.343, 5.890, 0.0]),
+        [np.nan, None, np.nan, None],
+        {"a": [1.123, 2.343, np.nan, np.nan], "b": [None, 3, 9.08, None]},
+    ],
+)
+@pytest.mark.parametrize("periods", range(-4, 5))
```

Comment:
Int, Float is already covered. Other dtypes should be added as well. bool, decimal type.
Specifically decimal type with different scale.

Reply:
@karthikeyann @brandon-b-miller  For some reason I'm getting this error with `Decimal32Dtype ` . Whereas
`Decimal64Dtype` passes, Is this expected behavior? 

```
E       TypeError: Operation sub not supported between dtypes decimal32 and decimal32.

python/cudf/cudf/core/column/column.py:1141: TypeError
```

Reply:
Hmm, that might be a bug somewhere in `DecimalColumn`. Certainly we should be able to do basic arithmetic with decimal columns. Can you paste the full error?

Reply:
It looks like the [`binary_operator` method is only defined for Decimal64 columns](https://github.com/rapidsai/cudf/blob/branch-22.02/python/cudf/cudf/core/column/decimal.py#L159). The 32-bit decimal dtype was added in #8438, but only the arrow-related methods were made available. I don't know what the rationale was, but for this PR to work for Decimal32 we'll need to also move `binary_operator` into `DecimalBaseColumn`.

Reply:
Thanks for the clarification@vyasr 👍🏾